async-opcua-codegen/src/nodeset/gen.rs

```diff
@@ -68,15 +71,25 @@ impl<'a> NodeSetCodeGenerator<'a> {
             .or_else(|| options.first())
     }
 
+    fn render_default_encoding_id(&self, node_id: &NodeId) -> Result<TokenStream, CodeGenError> {
+        let binary_id = ParsedNodeId::parse(&node_id.0)
+            .ok()
```

Comment:
```suggestion
        let binary_id = ParsedNodeId::parse(&node_id.0)?
```

just fail early here, the NodeId isn't going to be invalid, but might as well fail with the correct error if it somehow is.